deepspeed/runtime/data_pipeline/curriculum_scheduler.py

```diff
@@ -3,22 +3,28 @@
 '''
 import math
 from deepspeed.utils import logger
+from .constants import *
 
 
 class CurriculumScheduler(object):
     def __init__(self, config):
         super().__init__()
         self.state = {}
-        assert "curriculum_type" in config, "Curriculum learning requires the config 'curriculum_type'"
-        assert "min_difficulty" in config, "Curriculum learning requires the config 'min_difficulty'"
-        assert "max_difficulty" in config, "Curriculum learning requires the config 'max_difficulty'"
-        assert "schedule_type" in config, "Curriculum learning requires the config 'schedule_type'"
-        self.state['min_difficulty'] = config['min_difficulty']
-        self.state['max_difficulty'] = config['max_difficulty']
-        self.state['current_difficulty'] = config['min_difficulty']
-        self.state['schedule_type'] = config['schedule_type']
+        assert CURRICULUM_LEARNING_MIN_DIFFICULTY in config, \
+            f"Curriculum learning requires the config '{CURRICULUM_LEARNING_MIN_DIFFICULTY}'"
+        assert CURRICULUM_LEARNING_MAX_DIFFICULTY in config, \
+            f"Curriculum learning requires the config '{CURRICULUM_LEARNING_MAX_DIFFICULTY}'"
+        assert CURRICULUM_LEARNING_SCHEDULE_TYPE in config, \
+            f"Curriculum learning requires the config '{CURRICULUM_LEARNING_SCHEDULE_TYPE}'"
+        self.state[CURRICULUM_LEARNING_MIN_DIFFICULTY] = config[
+            CURRICULUM_LEARNING_MIN_DIFFICULTY]
+        self.state[CURRICULUM_LEARNING_MAX_DIFFICULTY] = config[
+            CURRICULUM_LEARNING_MAX_DIFFICULTY]
+        self.state['current_difficulty'] = config[CURRICULUM_LEARNING_MIN_DIFFICULTY]
```

Comment:
Replace `'current_difficulty'` with symbolic constant.  

Same for `'fixed_discrete'`, `'schedule'`, `'fixed_root'`, `'fixed_linear'`, and `'custom'`.

Reply:
Have changed them to symbolic constants.